libs/cohere/langchain_cohere/chat_models.py

```diff
@@ -410,6 +410,8 @@ def _get_generation_info(self, response: NonStreamedChatResponse) -> Dict[str, A
             )
         if hasattr(response, "token_count"):
             generation_info["token_count"] = response.token_count
+        elif hasattr(response.meta, "tokens") and response.meta.tokens is not None:
+            generation_info["token_count"] = response.meta.tokens.dict()
```

Comment:
supporting multiple Cohere SDK versions is a little tricker than I thought - here's a suggestion that passes linting (I tried it locally with 3.11 and 3.8). 
```suggestion
        elif hasattr(response, "meta") and response.meta is not None:
            if hasattr(response.meta, "tokens") and response.meta.tokens is not None:
                generation_info["token_count"] = response.meta.tokens.dict()
```

We'll also need to type hint the dictionary on line 397:

`generation_info: Dict[str, Any] = {`